util/chunk/chunk_test.go

```diff
@@ -1083,3 +1083,87 @@ func benchmarkChunkGrow(t benchChunkGrowCase) func(b *testing.B) {
 		}
 	}
 }
+func (s *testChunkSuite) TestAppendRows(c *check.C) {
+	numCols := 6
+	numRows := 10
+	chk := newChunk(8, 8, 0, 0, 40, 0)
+	strFmt := "%d.12345"
+	for i := 0; i < numRows; i++ {
+		chk.AppendNull(0)
+		chk.AppendInt64(1, int64(i))
+		str := fmt.Sprintf(strFmt, i)
+		chk.AppendString(2, str)
+		chk.AppendBytes(3, []byte(str))
+		chk.AppendMyDecimal(4, types.NewDecFromStringForTest(str))
+		chk.AppendJSON(5, json.CreateBinary(str))
+	}
+	c.Assert(chk.NumCols(), check.Equals, numCols)
+	c.Assert(chk.NumRows(), check.Equals, numRows)
+
+	chk2 := newChunk(8, 8, 0, 0, 40, 0)
+	c.Assert(chk.NumCols(), check.Equals, numCols)
+	rows := make([]Row, numRows)
+	for i := 0; i < numRows; i++ {
+		rows[i] = chk.GetRow(i)
+
+	}
+	chk2.AppendRows(rows)
+	for i := 0; i < numRows; i++ {
+		row := chk2.GetRow(i)
+		c.Assert(row.GetInt64(0), check.Equals, int64(0))
+		c.Assert(row.IsNull(0), check.IsTrue)
+		c.Assert(row.GetInt64(1), check.Equals, int64(i))
+		str := fmt.Sprintf(strFmt, i)
+		c.Assert(row.IsNull(2), check.IsFalse)
+		c.Assert(row.GetString(2), check.Equals, str)
+		c.Assert(row.IsNull(3), check.IsFalse)
+		c.Assert(row.GetBytes(3), check.BytesEquals, []byte(str))
+		c.Assert(row.IsNull(4), check.IsFalse)
+		c.Assert(row.GetMyDecimal(4).String(), check.Equals, str)
+		c.Assert(row.IsNull(5), check.IsFalse)
+		c.Assert(string(row.GetJSON(5).GetString()), check.Equals, str)
+	}
+}
+func BenchmarkBatchAppendRows(b *testing.B) {
```

Comment:
The two benchmarks are the same mostly so that I think use b.Run and transfer an arg to choose whether to use AppendRows or the old method is better. It can reuse the most code.